manifests/params.pp

```diff
@@ -16,272 +16,327 @@
 class snmp::params {
   # If we have a top scope variable defined, use it, otherwise fall back to a
   # hardcoded value.
-  $agentaddress = $::snmp_agentaddress ? {
-    undef   => [ 'udp:127.0.0.1:161', 'udp6:[::1]:161' ],
-    default => $::snmp_agentaddress,
+  if defined('$snmp_agentaddress') {
```

Comment:
I've not tried this bit it looks wrong. '$::variable' is not defined as it is a literal.

It should I believe be:

```puppet
if getvar('::snmp_agentaddress') {

}